types/wlr_surface.c

```diff
@@ -55,6 +82,7 @@ static void surface_commit(struct wl_client *client,
 			} else {
 				uint32_t format = wl_shm_buffer_get_format(buffer);
 				wlr_texture_upload_shm(surface->texture, format, buffer);
+				wl_resource_queue_event(surface->pending_buffer, WL_BUFFER_RELEASE);
```

Comment:
It might work. I'm following the example of weston here. They could be using old apis though.

Try grepping `wl_buffer_send_release` in the weston source and there are no results. Try grepping `WL_BUFFER_RELEASE` and you will see two results.